glob.go

```diff
@@ -336,15 +337,28 @@ func indexMatchedOpeningAlt(s string) int {
 }
 
 // Returns true if the path exists
-func (g *glob) exists(fsys fs.FS, name string) (bool, error) {
-	_, err := fs.Stat(fsys, name)
-	return err == nil, g.forwardErrIfFailOnIOErrors(err)
+func (g *glob) exists(fsys fs.FS, name string) (fs.FileInfo, bool, error) {
+	// name might end in a slash, but Stat doesn't like that
+	namelen := len(name)
+	if name[namelen - 1] == '/' {
```

Comment:
This needs to be

```go
if namelen > 1 && name[namelen-1] == '/' {
```

to handle the case where name is `/`, otherwise `fs.Stat` gets called with name being an empty string, which returns a permission error.

Reply:
Good catch! Thanks!